<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Quality Control of Sequencing Data • fastqcr</title>
<link rel="shortcut icon" type="image/x-icon" href="favicon.ico">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.1/clipboard.min.js" integrity="sha256-hIvIxeqhGZF+VVeM55k0mJvWpQ6gTkWk3Emc+NmowYA=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Quality Control of Sequencing Data">
<meta property="og:description" content="'FASTQC' is the most widely used tool for evaluating the quality of high throughput sequencing data.  
    It produces, for each sample, an html report and a compressed file containing the raw data. 
    If you have hundreds of samples, you are not going to open up each 'HTML' page. 
    You need some way of looking at these data in aggregate. 
    'fastqcr' Provides helper functions to easily parse, aggregate and analyze 
    'FastQC' reports for large numbers of samples. It provides a convenient solution for building 
    a 'Multi-QC' report, as well as, a 'one-sample' report with result interpretations.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">fastqcr</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    QC Reports
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="qc-reports/fastqcr-multi-qc-report.html">Multi-QC Reports</a>
    </li>
    <li>
      <a href="qc-reports/sample-qc-report-interpretation.html">One-Sample Report (+Interpretation)</a>
    </li>
    <li>
      <a href="qc-reports/sample-qc-report-without-interpretation.html">One-Sample Report (no interpretation)</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/kassambara/fastqcr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    

    
    
<!-- README.md is generated from README.Rmd. Please edit that file -->

<p><br></p>
<style>
.error,.notice,.warning,.success,.question{height:auto;padding:10px 10px 10px 40px;margin:5px auto 15px;line-height:20px;border:1px solid #FFF;border-radius:4px;position:relative;display:block;text-align:left}.question{background-color:#DAEEF8;border-color:#BDE9F2}.notice{background-color:#F0F0F0;border-color:#E2E2E2}.warning{background-color:#FDF9E4;border-color:#FBECCD}.error{background-color:#F3DFDF;border-color:#ECCDD2}.success{background-color:#E0F1D9;border-color:#D7EAC7}.block,.medium-block,.small-block{border:1px solid #CCC;border-top:2px solid #366393;border-bottom:1px solid #99B1CB;background:#F2F8FF;padding:10px}.block{width:auto;margin-top:10px;margin-bottom:10px}img{background-color:#fff;background-color:#FFF;border-radius:3px;border:1px solid #CCC;box-shadow:2px 2px 12px -5px #999;margin:0 5px;margin-bottom:5px;padding:5px;text-align:center}
</style>
<div id="fastqcr-quality-control-of-sequencing-data" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#fastqcr-quality-control-of-sequencing-data" class="anchor"></a>fastqcr: Quality Control of Sequencing Data</h1></div>
<p>The FastQC, written by Simon Andrews at the Babraham Institute, is the most widely used sequence quality assessment tool for evaluating the raw reads from high throughput sequencing data.</p>
<p>It produces, for each sample, an html report and a ‘zip’ file, which contains a file called fastqc_data.txt and summary.txt.</p>
<p>If you have hundreds of samples, you’re not going to open up each HTML page. You need some way of looking at these data in aggregate.</p>
<p>The <strong>fastqcr</strong> R package provides helper functions to easily and automatically parse, aggregate and analyze FastQC reports for large numbers of samples.</p>
<p>Additionally, the <strong>fastqcr</strong> package provides a convenient solution for building a multi-QC report and a one-sample FastQC report with the result interpretations. The online documentation is available at: <a href="http://www.sthda.com/english/rpkgs/fastqcr/" class="uri">http://www.sthda.com/english/rpkgs/fastqcr/</a>.</p>
<p>Examples of QC reports, generated automatically by the <strong>fastqcr</strong> R package, include:</p>
<ul>
<li><a href="http://www.sthda.com/english/rpkgs/fastqcr/qc-reports/fastqcr-multi-qc-report.html">Multi-QC report for multiple samples</a></li>
<li><a href="http://www.sthda.com/english/rpkgs/fastqcr/qc-reports/sample-qc-report-interpretation.html">One sample QC report (+ interpretation)</a></li>
<li><a href="http://www.sthda.com/english/rpkgs/fastqcr/qc-reports/sample-qc-report-without-interpretation.html">One sample QC report (no interpretation)</a></li>
</ul>
<div class="figure">
<img src="tools/fastqcr.png" alt="fastqcr logo"><p class="caption">fastqcr logo</p>
</div>
<div id="installation-and-loading" class="section level2">
<h2 class="hasAnchor">
<a href="#installation-and-loading" class="anchor"></a>Installation and loading</h2>
<ul>
<li>fastqcr can be installed from <a href="https://cran.r-project.org/package=fastqcr">CRAN</a> as follow:</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">install.packages</span>(<span class="st">"fastqcr"</span>)</code></pre></div>
<ul>
<li>Or, install the latest version from <a href="https://github.com/kassambara/fastqcr">GitHub</a>:</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>(<span class="op">!</span><span class="kw">require</span>(devtools)) <span class="kw">install.packages</span>(<span class="st">"devtools"</span>)
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="st">"kassambara/fastqcr"</span>)</code></pre></div>
<ul>
<li>Load fastqcr:</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"fastqcr"</span>)</code></pre></div>
</div>
<div id="quick-start" class="section level2">
<h2 class="hasAnchor">
<a href="#quick-start" class="anchor"></a>Quick Start</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(fastqcr)

<span class="co"># Aggregating Multiple FastQC Reports into a Data Frame </span>
<span class="co">#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%</span>

<span class="co"># Demo QC directory containing zipped FASTQC reports</span>
qc.dir &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/system.file">system.file</a></span>(<span class="st">"fastqc_results"</span>, <span class="dt">package =</span> <span class="st">"fastqcr"</span>)
qc &lt;-<span class="st"> </span><span class="kw"><a href="reference/qc_aggregate.html">qc_aggregate</a></span>(qc.dir)
qc

<span class="co"># Inspecting QC Problems</span>
<span class="co">#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%</span>

<span class="co"># See which modules failed in the most samples</span>
<span class="kw"><a href="reference/qc_problems.html">qc_fails</a></span>(qc, <span class="st">"module"</span>)
<span class="co"># Or, see which samples failed the most</span>
<span class="kw"><a href="reference/qc_problems.html">qc_fails</a></span>(qc, <span class="st">"sample"</span>)

<span class="co"># Building Multi QC Reports</span>
<span class="co">#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%</span>
<span class="kw"><a href="reference/qc_report.html">qc_report</a></span>(qc.dir, <span class="dt">result.file =</span> <span class="st">"multi-qc-report"</span> )

<span class="co"># Building One-Sample QC Reports (+ Interpretation)</span>
<span class="co">#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%</span>
qc.file &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/system.file">system.file</a></span>(<span class="st">"fastqc_results"</span>, <span class="st">"S1_fastqc.zip"</span>, <span class="dt">package =</span> <span class="st">"fastqcr"</span>)
<span class="kw"><a href="reference/qc_report.html">qc_report</a></span>(qc.file, <span class="dt">result.file =</span> <span class="st">"one-sample-report"</span>,
          <span class="dt">interpret =</span> <span class="ot">TRUE</span>)</code></pre></div>
</div>
<div id="main-functions" class="section level2">
<h2 class="hasAnchor">
<a href="#main-functions" class="anchor"></a>Main Functions</h2>
<p><strong>1) Installing and Running FastQC</strong></p>
<ul>
<li><p><strong>fastqc_install</strong>(): Install the latest version of FastQC tool on Unix systems (MAC OSX and Linux)</p></li>
<li><p><strong>fastqc</strong>(): Run the FastQC tool from R.</p></li>
</ul>
<p><strong>2) Aggregating and Summarizing Multiple FastQC Reports</strong></p>
<ul>
<li><p><strong>qc &lt;- qc_aggregate</strong>(): Aggregate multiple FastQC reports into a data frame.</p></li>
<li><p><strong>summary</strong>(qc): Generates a summary of qc_aggregate.</p></li>
<li><p><strong>qc_stats</strong>(qc): General statistics of FastQC reports.</p></li>
</ul>
<p><strong>3) Inspecting Problems</strong></p>
<ul>
<li><p><strong>qc_fails</strong>(qc): Displays samples or modules that failed.</p></li>
<li><p><strong>qc_warns</strong>(qc): Displays samples or modules that warned.</p></li>
<li><p><strong>qc_problems</strong>(qc): Union of <strong>qc_fails</strong>() and <strong>qc_warns</strong>(). Display which samples or modules that failed or warned.</p></li>
</ul>
<p><strong>4) Importing and Plotting FastQC Reports</strong></p>
<ul>
<li><p><strong>qc_read</strong>(): Read FastQC data into R.</p></li>
<li><p><strong>qc_plot</strong>(qc): Plot FastQC data</p></li>
</ul>
<p><strong>5) Building One-Sample and Multi-QC Reports</strong></p>
<ul>
<li>
<strong>qc_report</strong>(): Create an HTML file containing FastQC reports of one or multiple files. Inputs can be either a directory containing multiple FastQC reports or a single sample FastQC report.</li>
</ul>
<p><strong>6) Others</strong></p>
<ul>
<li>
<strong>qc_unzip</strong>(): Unzip all zipped files in the qc.dir directory. <br>
</li>
</ul>
</div>
<div id="installing-fastqc-from-r" class="section level2">
<h2 class="hasAnchor">
<a href="#installing-fastqc-from-r" class="anchor"></a>Installing FastQC from R</h2>
<p>You can install automatically the FastQC tool from R as follow:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/fastqc_install.html">fastqc_install</a></span>()</code></pre></div>
</div>
<div id="running-fastqc-from-r" class="section level2">
<h2 class="hasAnchor">
<a href="#running-fastqc-from-r" class="anchor"></a>Running FastQC from R</h2>
<p>The supported <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/2%20Basic%20Operations/2.1%20Opening%20a%20sequence%20file.html">file formats</a> by FastQC include:</p>
<ul>
<li>FASTQ</li>
<li>gzip compressed FASTQ</li>
</ul>
<p>Suppose that your working directory is organized as follow:</p>
<ul>
<li>home
<ul>
<li>Documents
<ul>
<li>FASTQ</li>
</ul>
</li>
</ul>
</li>
</ul>
<p>where, FASTQ is the directory containing your FASTQ files, for which you want to perform the quality control check.</p>
<p>To run FastQC from R, type this:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/fastqc.html">fastqc</a></span>(<span class="dt">fq.dir =</span> <span class="st">"~/Documents/FASTQ"</span>, <span class="co"># FASTQ files directory</span>
       <span class="dt">qc.dir =</span> <span class="st">"~/Documents/FASTQC"</span>, <span class="co"># Results direcory</span>
       <span class="dt">threads =</span> <span class="dv">4</span>                    <span class="co"># Number of threads</span>
       )</code></pre></div>
</div>
<div id="fastqc-reports" class="section level2">
<h2 class="hasAnchor">
<a href="#fastqc-reports" class="anchor"></a>FastQC Reports</h2>
<p>For each sample, FastQC performs a series of tests called <em>analysis modules</em>.</p>
<p>These modules include:</p>
<ul>
<li>Basic Statistics,</li>
<li>Per base sequence quality,</li>
<li>Per tile sequence quality</li>
<li>Per sequence quality scores,</li>
<li>Per base sequence content,</li>
<li>Per sequence GC content,</li>
<li>Per base N content,</li>
<li>Sequence Length Distribution,</li>
<li>Sequence Duplication Levels,</li>
<li>Overrepresented sequences,</li>
<li>Adapter Content</li>
<li>Kmer content</li>
</ul>
<p>The interpretation of these modules are provided in the official documentation of the <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/">FastQC tool</a>.</p>
</div>
<div id="aggregating-reports" class="section level2">
<h2 class="hasAnchor">
<a href="#aggregating-reports" class="anchor"></a>Aggregating Reports</h2>
<p>Here, we provide an R function <strong>qc_aggregate()</strong> to walk the FastQC result directory, find all the FASTQC zipped output folders, read the <strong>fastqc_data.txt</strong> and the <strong>summary.txt</strong> files, and aggregate the information into a data frame.</p>
<p>In the example below, we’ll use a demo FastQC output directory available in the fastqcr package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(fastqcr)
<span class="co"># Demo QC dir</span>
qc.dir &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/system.file">system.file</a></span>(<span class="st">"fastqc_results"</span>, <span class="dt">package =</span> <span class="st">"fastqcr"</span>)
qc.dir
<span class="co"># [1] "/Library/Frameworks/R.framework/Versions/3.3/Resources/library/fastqcr/fastqc_results"</span>
   
<span class="co"># List of files in the directory</span>
<span class="kw">list.files</span>(qc.dir)
<span class="co"># [1] "S1_fastqc.zip" "S2_fastqc.zip" "S3_fastqc.zip" "S4_fastqc.zip"</span>
<span class="co"># [5] "S5_fastqc.zip"</span></code></pre></div>
<p>The demo QC directory contains five zipped folders corresponding to the FastQC output for 5 samples.</p>
<p>Aggregating FastQC reports:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">qc &lt;-<span class="st"> </span><span class="kw"><a href="reference/qc_aggregate.html">qc_aggregate</a></span>(qc.dir)
qc</code></pre></div>
<p>The aggregated report looks like this:</p>
<table class="table table">
<thead><tr class="header">
<th align="left">sample</th>
<th align="left">module</th>
<th align="left">status</th>
<th align="left">tot.seq</th>
<th align="left">seq.length</th>
<th align="right">pct.gc</th>
<th align="right">pct.dup</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">S1</td>
<td align="left">Per tile sequence quality</td>
<td align="left">PASS</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">17.24</td>
</tr>
<tr class="even">
<td align="left">S4</td>
<td align="left">Sequence Duplication Levels</td>
<td align="left">PASS</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">19.89</td>
</tr>
<tr class="odd">
<td align="left">S2</td>
<td align="left">Per sequence quality scores</td>
<td align="left">PASS</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">15.70</td>
</tr>
<tr class="even">
<td align="left">S1</td>
<td align="left">Basic Statistics</td>
<td align="left">PASS</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">17.24</td>
</tr>
<tr class="odd">
<td align="left">S4</td>
<td align="left">Basic Statistics</td>
<td align="left">PASS</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">19.89</td>
</tr>
<tr class="even">
<td align="left">S3</td>
<td align="left">Per base sequence content</td>
<td align="left">FAIL</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">22.14</td>
</tr>
<tr class="odd">
<td align="left">S3</td>
<td align="left">Adapter Content</td>
<td align="left">PASS</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">22.14</td>
</tr>
<tr class="even">
<td align="left">S3</td>
<td align="left">Per sequence GC content</td>
<td align="left">FAIL</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">22.14</td>
</tr>
<tr class="odd">
<td align="left">S2</td>
<td align="left">Overrepresented sequences</td>
<td align="left">PASS</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">15.70</td>
</tr>
<tr class="even">
<td align="left">S2</td>
<td align="left">Per base sequence content</td>
<td align="left">FAIL</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">15.70</td>
</tr>
</tbody>
</table>
<p>Column names:</p>
<ul>
<li>
<strong>sample</strong>: sample names</li>
<li>
<strong>module</strong>: fastqc modules</li>
<li>
<strong>status</strong>: fastqc module status for each sample</li>
<li>
<strong>tot.seq</strong>: total sequences (i.e.: the number of reads)</li>
<li>
<strong>seq.length</strong>: sequence length</li>
<li>
<strong>pct.gc</strong>: percentage of GC content</li>
<li>
<strong>pct.dup</strong>: percentage of duplicate reads</li>
</ul>
<div class="block">
<p>
The table shows, for each sample, the names of tested FastQC modules, the status of the test, as well as, some general statistics including the number of reads, the length of reads, the percentage of GC content and the percentage of duplicate reads.
</p>
</div>
<p>Once you have the aggregated data you can use the <strong>dplyr</strong> package to easily inspect modules that failed or warned in samples. For example, the following R code shows samples with warnings and/or failures:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(dplyr)
qc <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw"><a href="http://dplyr.tidyverse.org/reference/select.html">select</a></span>(sample, module, status) <span class="op">%&gt;%</span><span class="st">    </span>
<span class="st">  </span><span class="kw"><a href="http://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(status <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">"WARN"</span>, <span class="st">"FAIL"</span>)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw"><a href="http://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span>(sample)
<span class="co"># # A tibble: 15 x 3</span>
<span class="co">#    sample module                       status</span>
<span class="co">#    &lt;chr&gt;  &lt;chr&gt;                        &lt;chr&gt; </span>
<span class="co">#  1 S1     Per base sequence content    FAIL  </span>
<span class="co">#  2 S1     Per sequence GC content      WARN  </span>
<span class="co">#  3 S1     Sequence Length Distribution WARN  </span>
<span class="co">#  4 S2     Per base sequence content    FAIL  </span>
<span class="co">#  5 S2     Per sequence GC content      WARN  </span>
<span class="co">#  6 S2     Sequence Length Distribution WARN  </span>
<span class="co">#  7 S3     Per base sequence content    FAIL  </span>
<span class="co">#  8 S3     Per sequence GC content      FAIL  </span>
<span class="co">#  9 S3     Sequence Length Distribution WARN  </span>
<span class="co"># 10 S4     Per base sequence content    FAIL  </span>
<span class="co"># 11 S4     Per sequence GC content      FAIL  </span>
<span class="co"># 12 S4     Sequence Length Distribution WARN  </span>
<span class="co"># 13 S5     Per base sequence content    FAIL  </span>
<span class="co"># 14 S5     Per sequence GC content      WARN  </span>
<span class="co"># 15 S5     Sequence Length Distribution WARN</span></code></pre></div>
<div class="success">
<p>
In the next section, we’ll describe some easy-to-use functions, available in the <strong>fastqcr</strong> package, for analyzing the aggregated data.
</p>
</div>
</div>
<div id="summarizing-reports" class="section level2">
<h2 class="hasAnchor">
<a href="#summarizing-reports" class="anchor"></a>Summarizing Reports</h2>
<p>We start by presenting a summary and general statistics of the aggregated data.</p>
<div id="qc-summary" class="section level3">
<h3 class="hasAnchor">
<a href="#qc-summary" class="anchor"></a>QC Summary</h3>
<ul>
<li>R function: <strong>summary</strong>()<br>
</li>
<li>Input data: aggregated data from <strong>qc_aggregate</strong>()</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Summary of qc</span>
<span class="kw">summary</span>(qc)
<span class="co"># # A tibble: 12 x 7</span>
<span class="co"># # Groups:   module [?]</span>
<span class="co">#    module         nb_samples nb_fail nb_pass nb_warn failed     warned    </span>
<span class="co">#    &lt;chr&gt;               &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;     </span>
<span class="co">#  1 Adapter Conte…          5       0       5       0 &lt;NA&gt;       &lt;NA&gt;      </span>
<span class="co">#  2 Basic Statist…          5       0       5       0 &lt;NA&gt;       &lt;NA&gt;      </span>
<span class="co">#  3 Kmer Content            5       0       5       0 &lt;NA&gt;       &lt;NA&gt;      </span>
<span class="co">#  4 Overrepresent…          5       0       5       0 &lt;NA&gt;       &lt;NA&gt;      </span>
<span class="co">#  5 Per base N co…          5       0       5       0 &lt;NA&gt;       &lt;NA&gt;      </span>
<span class="co">#  6 Per base sequ…          5       5       0       0 S1, S2, S… &lt;NA&gt;      </span>
<span class="co">#  7 Per base sequ…          5       0       5       0 &lt;NA&gt;       &lt;NA&gt;      </span>
<span class="co">#  8 Per sequence …          5       2       0       3 S3, S4     S1, S2, S5</span>
<span class="co">#  9 Per sequence …          5       0       5       0 &lt;NA&gt;       &lt;NA&gt;      </span>
<span class="co"># 10 Per tile sequ…          5       0       5       0 &lt;NA&gt;       &lt;NA&gt;      </span>
<span class="co"># 11 Sequence Dupl…          5       0       5       0 &lt;NA&gt;       &lt;NA&gt;      </span>
<span class="co"># 12 Sequence Leng…          5       0       0       5 &lt;NA&gt;       S1, S2, S…</span></code></pre></div>
<p>Column names:</p>
<ul>
<li>
<em>module</em>: fastqc modules</li>
<li>
<em>nb_samples</em>: the number of samples tested</li>
<li>
<em>nb_pass, nb_fail, nb_warn</em>: the number of samples that passed, failed and warned, respectively.</li>
<li>
<em>failed, warned</em>: the name of samples that failed and warned, respectively.</li>
</ul>
<div class="block">
<p>
The table shows, for each FastQC module, the number and the name of samples that failed or warned.
</p>
</div>
</div>
<div id="general-statistics" class="section level3">
<h3 class="hasAnchor">
<a href="#general-statistics" class="anchor"></a>General statistics</h3>
<ul>
<li>R function: <strong>qc_stats</strong>()</li>
<li>Input data: aggregated data from <strong>qc_aggregate</strong>()</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/qc_aggregate.html">qc_stats</a></span>(qc)
<span class="co"># # A tibble: 5 x 5</span>
<span class="co">#   sample pct.dup pct.gc tot.seq seq.length</span>
<span class="co">#   &lt;chr&gt;    &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;     </span>
<span class="co"># 1 S1        17.2     NA &lt;NA&gt;    &lt;NA&gt;      </span>
<span class="co"># 2 S2        15.7     NA &lt;NA&gt;    &lt;NA&gt;      </span>
<span class="co"># 3 S3        22.1     NA &lt;NA&gt;    &lt;NA&gt;      </span>
<span class="co"># 4 S4        19.9     NA &lt;NA&gt;    &lt;NA&gt;      </span>
<span class="co"># 5 S5        18.2     NA &lt;NA&gt;    &lt;NA&gt;</span></code></pre></div>
<p>Column names:</p>
<ul>
<li>
<em>pct.dup</em>: the percentage of duplicate reads,</li>
<li>
<em>pct.gc</em>: the percentage of GC content,</li>
<li>
<em>tot.seq</em>: total sequences or the number of reads and</li>
<li>
<em>seq.length</em>: sequence length or the length of reads.</li>
</ul>
<div class="block">
<p>
The table shows, for each sample, some general statistics such as the total number of reads, the length of reads, the percentage of GC content and the percentage of duplicate reads
</p>
</div>
</div>
</div>
<div id="inspecting-problems" class="section level2">
<h2 class="hasAnchor">
<a href="#inspecting-problems" class="anchor"></a>Inspecting Problems</h2>
<p>Once you’ve got this aggregated data, it’s easy to figure out what (if anything) is wrong with your data.</p>
<p><strong>1) R functions</strong>. You can inspect problems per either modules or samples using the following R functions:</p>
<ul>
<li>
<strong>qc_fails</strong>(qc): Displays samples or modules that failed.</li>
<li>
<strong>qc_warns</strong>(qc): Displays samples or modules that warned.</li>
<li>
<strong>qc_problems</strong>(qc): Union of <strong>qc_fails</strong>() and <strong>qc_warns</strong>(). Display which samples or modules that failed or warned.</li>
</ul>
<p><strong>2) Input data</strong>: aggregated data from <strong>qc_aggregate</strong>()</p>
<p><strong>3) Output data</strong>: Returns samples or FastQC modules with failures or warnings. By default, these functions return a compact output format. If you want a stretched format, specify the argument <em>compact = FALSE</em>.</p>
<p>The format and the interpretation of the outputs depend on the additional argument <em>element</em>, which value is one of c(“sample”, “module”).</p>
<ul>
<li>If <strong>element = “sample”</strong> (default), results are samples with failed and/or warned modules. The results contain the following columns:
<ul>
<li>sample (sample names),</li>
<li>nb_problems (the number of modules with problems),</li>
<li>module (the name of modules with problems).</li>
</ul>
</li>
<li>If <strong>element = “module”</strong>, results are modules that failed and/or warned in the most samples. The results contain the following columns:
<ul>
<li>module (the name of module with problems),</li>
<li>nb_problems (the number of samples with problems),</li>
<li>sample (the name of samples with problems)</li>
</ul>
</li>
</ul>
<div id="per-module-problems" class="section level3">
<h3 class="hasAnchor">
<a href="#per-module-problems" class="anchor"></a>Per Module Problems</h3>
<ul>
<li>
<strong>Modules that failed in the most samples</strong>:</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># See which module failed in the most samples</span>
<span class="kw"><a href="reference/qc_problems.html">qc_fails</a></span>(qc, <span class="st">"module"</span>)
<span class="co"># # A tibble: 2 x 3</span>
<span class="co">#   module                    nb_problems sample            </span>
<span class="co">#   &lt;chr&gt;                           &lt;int&gt; &lt;chr&gt;             </span>
<span class="co"># 1 Per base sequence content           5 S1, S2, S3, S4, S5</span>
<span class="co"># 2 Per sequence GC content             2 S3, S4</span></code></pre></div>
<div class="success">
<p>
For each module, the number of problems (failures) and the name of samples, that failed, are shown.
</p>
</div>
<ul>
<li>
<strong>Modules that warned in the most samples</strong>:</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># See which module warned in the most samples</span>
<span class="kw"><a href="reference/qc_problems.html">qc_warns</a></span>(qc, <span class="st">"module"</span>)
<span class="co"># # A tibble: 2 x 3</span>
<span class="co">#   module                       nb_problems sample            </span>
<span class="co">#   &lt;chr&gt;                              &lt;int&gt; &lt;chr&gt;             </span>
<span class="co"># 1 Sequence Length Distribution           5 S1, S2, S3, S4, S5</span>
<span class="co"># 2 Per sequence GC content                3 S1, S2, S5</span></code></pre></div>
<ul>
<li>
<strong>Modules that failed or warned</strong>: Union of qc_fails() and qc_warns()</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># See which modules failed or warned.</span>
<span class="kw"><a href="reference/qc_problems.html">qc_problems</a></span>(qc, <span class="st">"module"</span>)
<span class="co"># # A tibble: 3 x 3</span>
<span class="co">#   module                       nb_problems sample            </span>
<span class="co">#   &lt;chr&gt;                              &lt;int&gt; &lt;chr&gt;             </span>
<span class="co"># 1 Per base sequence content              5 S1, S2, S3, S4, S5</span>
<span class="co"># 2 Per sequence GC content                5 S1, S2, S3, S4, S5</span>
<span class="co"># 3 Sequence Length Distribution           5 S1, S2, S3, S4, S5</span></code></pre></div>
<p>The output above is in a compact format. For a stretched format, type this:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/qc_problems.html">qc_problems</a></span>(qc, <span class="st">"module"</span>, <span class="dt">compact =</span> <span class="ot">FALSE</span>)
<span class="co"># # A tibble: 15 x 4</span>
<span class="co">#    module                       nb_problems sample status</span>
<span class="co">#    &lt;chr&gt;                              &lt;int&gt; &lt;chr&gt;  &lt;chr&gt; </span>
<span class="co">#  1 Per base sequence content              5 S1     FAIL  </span>
<span class="co">#  2 Per base sequence content              5 S2     FAIL  </span>
<span class="co">#  3 Per base sequence content              5 S3     FAIL  </span>
<span class="co">#  4 Per base sequence content              5 S4     FAIL  </span>
<span class="co">#  5 Per base sequence content              5 S5     FAIL  </span>
<span class="co">#  6 Per sequence GC content                5 S3     FAIL  </span>
<span class="co">#  7 Per sequence GC content                5 S4     FAIL  </span>
<span class="co">#  8 Per sequence GC content                5 S1     WARN  </span>
<span class="co">#  9 Per sequence GC content                5 S2     WARN  </span>
<span class="co"># 10 Per sequence GC content                5 S5     WARN  </span>
<span class="co"># 11 Sequence Length Distribution           5 S1     WARN  </span>
<span class="co"># 12 Sequence Length Distribution           5 S2     WARN  </span>
<span class="co"># 13 Sequence Length Distribution           5 S3     WARN  </span>
<span class="co"># 14 Sequence Length Distribution           5 S4     WARN  </span>
<span class="co"># 15 Sequence Length Distribution           5 S5     WARN</span></code></pre></div>
<div class="success">
<p>
In the the stretched format each row correspond to a unique sample. Additionally, the status of each module is specified.
</p>
</div>
<p>It’s also possible to display problems for one or more specified modules. For example,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/qc_problems.html">qc_problems</a></span>(qc, <span class="st">"module"</span>,  <span class="dt">name =</span> <span class="st">"Per sequence GC content"</span>)
<span class="co"># # A tibble: 5 x 4</span>
<span class="co">#   module                  nb_problems sample status</span>
<span class="co">#   &lt;chr&gt;                         &lt;int&gt; &lt;chr&gt;  &lt;chr&gt; </span>
<span class="co"># 1 Per sequence GC content           5 S3     FAIL  </span>
<span class="co"># 2 Per sequence GC content           5 S4     FAIL  </span>
<span class="co"># 3 Per sequence GC content           5 S1     WARN  </span>
<span class="co"># 4 Per sequence GC content           5 S2     WARN  </span>
<span class="co"># 5 Per sequence GC content           5 S5     WARN</span></code></pre></div>
<div class="warning">
<p>
Note that, partial matching of name is allowed. For example, name = “Per sequence GC content” equates to name = “GC content”.
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/qc_problems.html">qc_problems</a></span>(qc, <span class="st">"module"</span>,  <span class="dt">name =</span> <span class="st">"GC content"</span>)</code></pre></div>
</div>
<div id="per-sample-problems" class="section level3">
<h3 class="hasAnchor">
<a href="#per-sample-problems" class="anchor"></a>Per Sample Problems</h3>
<ul>
<li><strong>Samples with one or more failed modules</strong></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># See which samples had one or more failed modules</span>
<span class="kw"><a href="reference/qc_problems.html">qc_fails</a></span>(qc, <span class="st">"sample"</span>)
<span class="co"># # A tibble: 5 x 3</span>
<span class="co">#   sample nb_problems module                                            </span>
<span class="co">#   &lt;chr&gt;        &lt;int&gt; &lt;chr&gt;                                             </span>
<span class="co"># 1 S3               2 Per base sequence content, Per sequence GC content</span>
<span class="co"># 2 S4               2 Per base sequence content, Per sequence GC content</span>
<span class="co"># 3 S1               1 Per base sequence content                         </span>
<span class="co"># 4 S2               1 Per base sequence content                         </span>
<span class="co"># 5 S5               1 Per base sequence content</span></code></pre></div>
<div class="success">
<p>
For each sample, the number of problems (failures) and the name of modules, that failed, are shown.
</p>
</div>
<ul>
<li>
<strong>Samples with failed or warned modules</strong>:</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># See which samples had one or more module with failure or warning</span>
<span class="kw"><a href="reference/qc_problems.html">qc_problems</a></span>(qc, <span class="st">"sample"</span>, <span class="dt">compact =</span> <span class="ot">FALSE</span>)
<span class="co"># # A tibble: 15 x 4</span>
<span class="co">#    sample nb_problems module                       status</span>
<span class="co">#    &lt;chr&gt;        &lt;int&gt; &lt;chr&gt;                        &lt;chr&gt; </span>
<span class="co">#  1 S1               3 Per base sequence content    FAIL  </span>
<span class="co">#  2 S1               3 Per sequence GC content      WARN  </span>
<span class="co">#  3 S1               3 Sequence Length Distribution WARN  </span>
<span class="co">#  4 S2               3 Per base sequence content    FAIL  </span>
<span class="co">#  5 S2               3 Per sequence GC content      WARN  </span>
<span class="co">#  6 S2               3 Sequence Length Distribution WARN  </span>
<span class="co">#  7 S3               3 Per base sequence content    FAIL  </span>
<span class="co">#  8 S3               3 Per sequence GC content      FAIL  </span>
<span class="co">#  9 S3               3 Sequence Length Distribution WARN  </span>
<span class="co"># 10 S4               3 Per base sequence content    FAIL  </span>
<span class="co"># 11 S4               3 Per sequence GC content      FAIL  </span>
<span class="co"># 12 S4               3 Sequence Length Distribution WARN  </span>
<span class="co"># 13 S5               3 Per base sequence content    FAIL  </span>
<span class="co"># 14 S5               3 Per sequence GC content      WARN  </span>
<span class="co"># 15 S5               3 Sequence Length Distribution WARN</span></code></pre></div>
<p>To specify the name of a sample of interest, type this:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/qc_problems.html">qc_problems</a></span>(qc, <span class="st">"sample"</span>, <span class="dt">name =</span> <span class="st">"S1"</span>)
<span class="co"># # A tibble: 3 x 4</span>
<span class="co">#   sample nb_problems module                       status</span>
<span class="co">#   &lt;chr&gt;        &lt;int&gt; &lt;chr&gt;                        &lt;chr&gt; </span>
<span class="co"># 1 S1               3 Per base sequence content    FAIL  </span>
<span class="co"># 2 S1               3 Per sequence GC content      WARN  </span>
<span class="co"># 3 S1               3 Sequence Length Distribution WARN</span></code></pre></div>
</div>
</div>
<div id="building-an-html-report" class="section level2">
<h2 class="hasAnchor">
<a href="#building-an-html-report" class="anchor"></a>Building an HTML Report</h2>
<p>The function <strong>qc_report</strong>() can be used to build a report of FastQC outputs. It creates an HTML file containing FastQC reports of one or multiple samples.</p>
<p>Inputs can be either a directory containing multiple FastQC reports or a single sample FastQC report.</p>
<div id="create-a-multi-qc-report" class="section level3">
<h3 class="hasAnchor">
<a href="#create-a-multi-qc-report" class="anchor"></a>Create a Multi-QC Report</h3>
<p>We’ll build a multi-qc report for the following demo QC directory:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Demo QC Directory</span>
qc.dir &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/system.file">system.file</a></span>(<span class="st">"fastqc_results"</span>, <span class="dt">package =</span> <span class="st">"fastqcr"</span>)
qc.dir
<span class="co"># [1] "/Library/Frameworks/R.framework/Versions/3.3/Resources/library/fastqcr/fastqc_results"</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Build a report</span>
<span class="kw"><a href="reference/qc_report.html">qc_report</a></span>(qc.dir, <span class="dt">result.file =</span> <span class="st">"~/Desktop/multi-qc-result"</span>,
          <span class="dt">experiment =</span> <span class="st">"Exome sequencing of colon cancer cell lines"</span>)</code></pre></div>
<div class="success">
<p>
An example of report is available at: <a href="http://www.sthda.com/english/rpkgs/fastqcr/qc-reports/fastqcr-multi-qc-report.html" target="_blank"> fastqcr multi-qc report</a>
</p>
</div>
</div>
<div id="create-a-one-sample-report" class="section level3">
<h3 class="hasAnchor">
<a href="#create-a-one-sample-report" class="anchor"></a>Create a One-Sample Report</h3>
<p>We’ll build a report for the following demo QC file:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"> qc.file &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/system.file">system.file</a></span>(<span class="st">"fastqc_results"</span>, <span class="st">"S1_fastqc.zip"</span>, <span class="dt">package =</span> <span class="st">"fastqcr"</span>)
qc.file
<span class="co"># [1] "/Library/Frameworks/R.framework/Versions/3.3/Resources/library/fastqcr/fastqc_results/S1_fastqc.zip"</span></code></pre></div>
<ul>
<li>
<strong>One-Sample QC report with plot interpretations</strong>:</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"> <span class="kw"><a href="reference/qc_report.html">qc_report</a></span>(qc.file, <span class="dt">result.file =</span> <span class="st">"one-sample-report-with-interpretation"</span>,
   <span class="dt">interpret =</span> <span class="ot">TRUE</span>)</code></pre></div>
<div class="success">
<p>
An example of report is available at: <a href="http://www.sthda.com/english/rpkgs/fastqcr/qc-reports/sample-qc-report-interpretation.html" target="_blank"> One sample QC report with interpretation</a>
</p>
</div>
<ul>
<li>
<strong>One-Sample QC report without plot interpretations</strong>:</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"> <span class="kw"><a href="reference/qc_report.html">qc_report</a></span>(qc.file, <span class="dt">result.file =</span> <span class="st">"one-sample-report"</span>,
   <span class="dt">interpret =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div class="success">
<p>
An example of report is available at: <a href="http://www.sthda.com/english/rpkgs/fastqcr/qc-reports/sample-qc-report-without-interpretation.html" target="_blank"> One sample QC report without interpretation</a>
</p>
</div>
</div>
</div>
<div id="importing-and-plotting-a-fastqc-qc-report" class="section level2">
<h2 class="hasAnchor">
<a href="#importing-and-plotting-a-fastqc-qc-report" class="anchor"></a>Importing and Plotting a FastQC QC Report</h2>
<p>We’ll visualize the output for sample 1:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Demo file</span>
qc.file &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/system.file">system.file</a></span>(<span class="st">"fastqc_results"</span>, <span class="st">"S1_fastqc.zip"</span>,  <span class="dt">package =</span> <span class="st">"fastqcr"</span>)
qc.file
<span class="co"># [1] "/Library/Frameworks/R.framework/Versions/3.3/Resources/library/fastqcr/fastqc_results/S1_fastqc.zip"</span></code></pre></div>
<p>We start by reading the output using the function <strong>qc_read</strong>(), which returns a list of tibbles containing the data for specified modules:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Read all modules</span>
qc &lt;-<span class="st"> </span><span class="kw"><a href="reference/qc_read.html">qc_read</a></span>(qc.file)
<span class="co"># Elements contained in the qc object</span>
<span class="kw">names</span>(qc)
<span class="co">#  [1] "summary"                       "basic_statistics"             </span>
<span class="co">#  [3] "per_base_sequence_quality"     "per_tile_sequence_quality"    </span>
<span class="co">#  [5] "per_sequence_quality_scores"   "per_base_sequence_content"    </span>
<span class="co">#  [7] "per_sequence_gc_content"       "per_base_n_content"           </span>
<span class="co">#  [9] "sequence_length_distribution"  "sequence_duplication_levels"  </span>
<span class="co"># [11] "overrepresented_sequences"     "adapter_content"              </span>
<span class="co"># [13] "kmer_content"                  "total_deduplicated_percentage"</span></code></pre></div>
<p>The function <strong>qc_plot</strong>() is used to visualized the data of a specified module. Allowed values for the argument modules include one or the combination of:</p>
<ul>
<li>“Summary”,</li>
<li>“Basic Statistics”,</li>
<li>“Per base sequence quality”,</li>
<li>“Per sequence quality scores”,</li>
<li>“Per base sequence content”,</li>
<li>“Per sequence GC content”,</li>
<li>“Per base N content”,</li>
<li>“Sequence Length Distribution”,</li>
<li>“Sequence Duplication Levels”,</li>
<li>“Overrepresented sequences”,</li>
<li>“Adapter Content”</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/qc_plot.html">qc_plot</a></span>(qc, <span class="st">"Per sequence GC content"</span>)

<span class="kw"><a href="reference/qc_plot.html">qc_plot</a></span>(qc, <span class="st">"Per base sequence quality"</span>)

<span class="kw"><a href="reference/qc_plot.html">qc_plot</a></span>(qc, <span class="st">"Per sequence quality scores"</span>)

<span class="kw"><a href="reference/qc_plot.html">qc_plot</a></span>(qc, <span class="st">"Per base sequence content"</span>)

<span class="kw"><a href="reference/qc_plot.html">qc_plot</a></span>(qc, <span class="st">"Sequence duplication levels"</span>)</code></pre></div>
<p><img src="tools/README-qc-plot-1.png" width="336"><img src="tools/README-qc-plot-2.png" width="336"><img src="tools/README-qc-plot-3.png" width="336"><img src="tools/README-qc-plot-4.png" width="336"><img src="tools/README-qc-plot-5.png" width="336"></p>
</div>
<div id="useful-links" class="section level2">
<h2 class="hasAnchor">
<a href="#useful-links" class="anchor"></a>Useful Links</h2>
<ul>
<li>FastQC report for a <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html">good Illumina dataset</a>
</li>
<li>FastQC report for a <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html">bad Illumina dataset</a>
</li>
<li><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/">Online documentation for each FastQC report</a></li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from CRAN at <br><a href="https://cloud.r-project.org/package=fastqcr">https://​cloud.r-project.org/​package=fastqcr</a>
</li>
<li>Report a bug at <br><a href="https://github.com/kassambara/fastqcr/issues">https://​github.com/​kassambara/​fastqcr/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Alboukadel Kassambara <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>

      <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.org/kassambara/fastqcr"><img src="https://api.travis-ci.org/kassambara/fastqcr.png" alt="Build Status"></a></li>
<li><a href="https://cran.r-project.org/package=fastqcr"><img src="http://www.r-pkg.org/badges/version/fastqcr" alt="CRAN_Status_Badge"></a></li>
<li><a href="https://cran.r-project.org/package=fastqcr"><img src="http://cranlogs.r-pkg.org/badges/fastqcr" alt="Downloads"></a></li>
<li><a href="http://cranlogs.r-pkg.org/badges/grand-total/fastqcr"><img src="http://cranlogs.r-pkg.org/badges/grand-total/fastqcr?color=orange" alt="Total Downloads"></a></li>
</ul>
</div>
</div>

</div>


      <footer><div class="copyright">
  <p>Developed by Alboukadel Kassambara.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
