<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Aggregate FastQC Reports for Multiple Samples â€” qc_aggregate â€¢ fastqcr</title>

<link rel="shortcut icon" type="image/x-icon" href="../favicon.ico">

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.1/clipboard.min.js" integrity="sha256-hIvIxeqhGZF+VVeM55k0mJvWpQ6gTkWk3Emc+NmowYA=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Aggregate FastQC Reports for Multiple Samples â€” qc_aggregate" />

<meta property="og:description" content="Aggregate multiple FastQC reports into a data frame." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">fastqcr</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    QC Reports
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../qc-reports/fastqcr-multi-qc-report.html">Multi-QC Reports</a>
    </li>
    <li>
      <a href="../qc-reports/sample-qc-report-interpretation.html">One-Sample Report (+Interpretation)</a>
    </li>
    <li>
      <a href="../qc-reports/sample-qc-report-without-interpretation.html">One-Sample Report (no interpretation)</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/kassambara/fastqcr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Aggregate FastQC Reports for Multiple Samples</h1>
    
    <div class="hidden name"><code>qc_aggregate.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Aggregate multiple FastQC reports into a data frame.</p>
    
    </div>

    <pre class="usage"><span class='fu'>qc_aggregate</span>(<span class='kw'>qc.dir</span> <span class='kw'>=</span> <span class='st'>"."</span>, <span class='kw'>progressbar</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)

<span class='co'># S3 method for qc_aggregate</span>
<span class='fu'>summary</span>(<span class='no'>object</span>, <span class='no'>...</span>)

<span class='fu'>qc_stats</span>(<span class='no'>object</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>qc.dir</th>
      <td><p>path to the FastQC result directory to scan.</p></td>
    </tr>
    <tr>
      <th>progressbar</th>
      <td><p>logical value. If TRUE, shows a progress bar.</p></td>
    </tr>
    <tr>
      <th>object</th>
      <td><p>an object of class qc_aggregate.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>other arguments.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    
<ul>
<li><p><strong>qc_aggregate()</strong> returns an object of class qc_aggregate
    which is a (tibble) data frame with the following column names:</p><ul>
<li><p>sample: sample names</p></li>
<li><p>module: fastqc modules</p></li>
<li><p>status: fastqc module status for each sample</p></li>
<li><p>tot.seq: total sequences (i.e.: the number of reads)</p></li>
<li><p>seq.length: sequence length</p></li>
<li><p>pct.gc: % of GC content</p></li>
<li><p>pct.dup: % of duplicate reads</p></li>
</ul></li>
<li><p><strong>summary</strong>: Generates a summary of qc_aggregate.
   Returns a data frame with the following columns:</p><ul>
<li><p>module: fastqc modules</p></li>
<li><p>nb_samples: the number of samples tested</p></li>
<li><p>nb_pass, nb_fail, nb_warn: the number of samples that passed, failed and warned, respectively.</p></li>
<li><p>failed, warned: the name of samples that failed and warned, respectively.</p></li>
</ul></li>
<li><p><strong>qc_stats</strong>: returns a data frame containing general statistics of fastqc reports.
   columns are: sample, pct.dup, pct.gc, tot.seq and seq.length.</p></li>
</ul>

    
    <h2 class="hasAnchor" id="functions"><a class="anchor" href="#functions"></a>Functions</h2>

    <ul>
<li><p><code>qc_aggregate</code>: Aggregate FastQC Reports for Multiple Samples</p></li>
<li><p><code>qc_stats</code>: Creates general statistics of fastqc reports.</p></li>
</ul>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Demo QC dir</span>
<span class='no'>qc.dir</span> <span class='kw'>&lt;-</span> <span class='fu'>system.file</span>(<span class='st'>"fastqc_results"</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"fastqcr"</span>)
<span class='no'>qc.dir</span></div><div class='output co'>#&gt; [1] "/Users/kassambara/Documents/R/MyPackages/fastqcr/inst/fastqc_results"</div><div class='input'>
<span class='co'># List of files in the directory</span>
<span class='fu'>list.files</span>(<span class='no'>qc.dir</span>)</div><div class='output co'>#&gt; [1] "S1_fastqc.zip" "S2_fastqc.zip" "S3_fastqc.zip" "S4_fastqc.zip"
#&gt; [5] "S5_fastqc.zip"</div><div class='input'>
<span class='co'># Aggregate the report</span>
<span class='no'>qc</span> <span class='kw'>&lt;-</span> <span class='fu'>qc_aggregate</span>(<span class='no'>qc.dir</span>, <span class='kw'>progressbar</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='no'>qc</span></div><div class='output co'>#&gt; [90m# A tibble: 60 x 7[39m
#&gt;    sample module                       status tot.seq seq.length pct.gc pct.dup
#&gt;  [90m*[39m [3m[90m&lt;chr&gt;[39m[23m  [3m[90m&lt;chr&gt;[39m[23m                        [3m[90m&lt;chr&gt;[39m[23m  [3m[90m&lt;chr&gt;[39m[23m   [3m[90m&lt;chr&gt;[39m[23m       [3m[90m&lt;dbl&gt;[39m[23m   [3m[90m&lt;dbl&gt;[39m[23m
#&gt; [90m 1[39m S1     Basic Statistics             PASS   [31mNA[39m      [31mNA[39m             [31mNA[39m    17.2
#&gt; [90m 2[39m S1     Per base sequence quality    PASS   [31mNA[39m      [31mNA[39m             [31mNA[39m    17.2
#&gt; [90m 3[39m S1     Per tile sequence quality    PASS   [31mNA[39m      [31mNA[39m             [31mNA[39m    17.2
#&gt; [90m 4[39m S1     Per sequence quality scores  PASS   [31mNA[39m      [31mNA[39m             [31mNA[39m    17.2
#&gt; [90m 5[39m S1     Per base sequence content    FAIL   [31mNA[39m      [31mNA[39m             [31mNA[39m    17.2
#&gt; [90m 6[39m S1     Per sequence GC content      WARN   [31mNA[39m      [31mNA[39m             [31mNA[39m    17.2
#&gt; [90m 7[39m S1     Per base N content           PASS   [31mNA[39m      [31mNA[39m             [31mNA[39m    17.2
#&gt; [90m 8[39m S1     Sequence Length Distribution WARN   [31mNA[39m      [31mNA[39m             [31mNA[39m    17.2
#&gt; [90m 9[39m S1     Sequence Duplication Levels  PASS   [31mNA[39m      [31mNA[39m             [31mNA[39m    17.2
#&gt; [90m10[39m S1     Overrepresented sequences    PASS   [31mNA[39m      [31mNA[39m             [31mNA[39m    17.2
#&gt; [90m# ... with 50 more rows[39m</div><div class='input'>
<span class='co'># Generates a summary of qc_aggregate</span>
<span class='fu'>summary</span>(<span class='no'>qc</span>)</div><div class='output co'>#&gt; [90m# A tibble: 12 x 7[39m
#&gt; [90m# Groups:   module [?][39m
#&gt;    module            nb_samples nb_fail nb_pass nb_warn failed      warned     
#&gt;    [3m[90m&lt;chr&gt;[39m[23m                  [3m[90m&lt;dbl&gt;[39m[23m   [3m[90m&lt;dbl&gt;[39m[23m   [3m[90m&lt;dbl&gt;[39m[23m   [3m[90m&lt;dbl&gt;[39m[23m [3m[90m&lt;chr&gt;[39m[23m       [3m[90m&lt;chr&gt;[39m[23m      
#&gt; [90m 1[39m Adapter Content            5       0       5       0 [31mNA[39m          [31mNA[39m         
#&gt; [90m 2[39m Basic Statistics           5       0       5       0 [31mNA[39m          [31mNA[39m         
#&gt; [90m 3[39m Kmer Content               5       0       5       0 [31mNA[39m          [31mNA[39m         
#&gt; [90m 4[39m Overrepresented â€¦          5       0       5       0 [31mNA[39m          [31mNA[39m         
#&gt; [90m 5[39m Per base N conteâ€¦          5       0       5       0 [31mNA[39m          [31mNA[39m         
#&gt; [90m 6[39m Per base sequencâ€¦          5       5       0       0 S1, S2, S3â€¦ [31mNA[39m         
#&gt; [90m 7[39m Per base sequencâ€¦          5       0       5       0 [31mNA[39m          [31mNA[39m         
#&gt; [90m 8[39m Per sequence GC â€¦          5       2       0       3 S3, S4      S1, S2, S5 
#&gt; [90m 9[39m Per sequence quaâ€¦          5       0       5       0 [31mNA[39m          [31mNA[39m         
#&gt; [90m10[39m Per tile sequencâ€¦          5       0       5       0 [31mNA[39m          [31mNA[39m         
#&gt; [90m11[39m Sequence Duplicaâ€¦          5       0       5       0 [31mNA[39m          [31mNA[39m         
#&gt; [90m12[39m Sequence Length â€¦          5       0       0       5 [31mNA[39m          S1, S2, S3â€¦</div><div class='input'>
<span class='co'># General statistics of fastqc reports.</span>
<span class='fu'>qc_stats</span>(<span class='no'>qc</span>)</div><div class='output co'>#&gt; [90m# A tibble: 5 x 5[39m
#&gt;   sample pct.dup pct.gc tot.seq seq.length
#&gt;   [3m[90m&lt;chr&gt;[39m[23m    [3m[90m&lt;dbl&gt;[39m[23m  [3m[90m&lt;dbl&gt;[39m[23m [3m[90m&lt;chr&gt;[39m[23m   [3m[90m&lt;chr&gt;[39m[23m     
#&gt; [90m1[39m S1        17.2     [31mNA[39m [31mNA[39m      [31mNA[39m        
#&gt; [90m2[39m S2        15.7     [31mNA[39m [31mNA[39m      [31mNA[39m        
#&gt; [90m3[39m S3        22.1     [31mNA[39m [31mNA[39m      [31mNA[39m        
#&gt; [90m4[39m S4        19.9     [31mNA[39m [31mNA[39m      [31mNA[39m        
#&gt; [90m5[39m S5        18.2     [31mNA[39m [31mNA[39m      [31mNA[39m        </div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#functions">Functions</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Alboukadel Kassambara.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

